nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -155,6 +154,11 @@ sub GetFs {
 
         # Is it better than our current match?
         if (length($mountPoint) > length($bestFs->mount)) {
+
+            # Check if it's a real directory. Do this last, because it does a stat(),
```

Comment:
Is this the only bottleneck? Looks like there are a few mktmp / tree walking operations throughout the script

but LGTM as an inconsequential change.

Reply:
> Is this the only bottleneck? Looks like there are a few mktmp / tree walking operations throughout the script

As far as I could tell, yes.

The tree walking operations wouldn't go into arbitrary mount points, but this stat did.